src/idpi/products/pot_vortic.py

```diff
@@ -0,0 +1,68 @@
+# Standard library
+import sys
+import logging
+
+# Third-party
+import numpy as np
+
+# First-party
+from idpi.operators.destagger import destagger
+from idpi.operators.pot_vortic import fpotvortic
+from idpi.operators.rho import f_rho_tot
+from idpi.operators.theta import ftheta
+from idpi.operators.total_diff import TotalDiff
+from idpi.operators.vertical_interpolation import interpolate_k2p
+from idpi.operators.vertical_interpolation import interpolate_k2theta
+from idpi.operators.vertical_reduction import integrate_k
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def _compute_pot_vortic(U, V, W, T, P, QV, QC, QI, HHL, theta):
+    logger.info("Computing total density")
+    rho_tot = f_rho_tot(T, P, QV, QC, QI)
+
+    logger.info("Computing terrain following grid deformation factors")
+    dlon = HHL.attrs["GRIB_iDirectionIncrementInDegrees"]
+    dlat = HHL.attrs["GRIB_jDirectionIncrementInDegrees"]
+    deg2rad = np.pi / 180
+    total_diff = TotalDiff(dlon * deg2rad, dlat * deg2rad, HHL)
+
+    logger.info("Computing potential vorticity")
+    return fpotvortic(U, V, W, theta, rho_tot, total_diff)
+
+
+def _compute_mean(pot_vortic, hhl, hfl, pressure):
+    logger.info("Computing mean potential vorticity between 700 and 900 hPa")
+    h700, h900 = interpolate_k2p(hfl, "linear_in_lnp", pressure, [700, 900], "hPa")
+    return integrate_k(pot_vortic, "normed_integral", "z2z", hhl, (h900, h700))
+
+
+def _compute_at_theta(theta, hfl, **fields):
+    logger.info(f"Interpolating {tuple(fields.keys())} at isotherms")
+    theta_values = [310, 315, 320, 325, 330, 335]
+    return {
+        key: interpolate_k2theta(field, "low_fold", theta, theta_values, "K", hfl)
+        for key, field in fields.items()
+    }
+
+
+def product2(U, V, W, T, P, QV, QC, QI, HHL):
```

Comment:
we should check with @petrabaumann if there is already an identifier/name proposed for the product, product2 is arbitrary

Reply:
I didn't have any other reference to use for a name

Reply:
I just checked with Petra. In the future we will have identifier for products that are disseminated through FabIO, but we do not have it yet. 
Petra suggested to get a name from the output filename of the Fieldextra product: 
https://github.com/MeteoSwiss-APN/oprconfig/blob/1661ea7595c257d28452e2093ff033844b1b40bb/nl_rendering/templates/c2e_fcst/products_c/ninjo_5/alpk2th_ctrl.nl#L39
In this case could be something like
```ninjo_cosmo_2e_ctrl_k2th```

Notice the filename contains `alpk2th`, which I guess refers to alp domain, which happens throught the index selection
https://github.com/MeteoSwiss-APN/oprconfig/blob/1661ea7595c257d28452e2093ff033844b1b40bb/nl_rendering/templates/c2e_fcst/products_c/ninjo_5/alpk2th_ctrl.nl#LL10C1-L11C1
But this product does not seem to implement that region selection, so that is why I woud not use  'alp' in the name

Reply:
I think the domain should be an input argument (part of the API) to the operator. That way we can have this product configured on any domain. A domain is a subset of the entire hypercube, including all dimensions like ensembles. 
So I think `ctrl` is also part of a domain specification `{ens=0}`
Therefore I would reduce the name to `ninjo_k2th`

Reply:
Okay renamed the symbols and files